src/ParallelEnv.f90

```diff
@@ -24,25 +24,38 @@ MODULE ParallelEnv
 USE Allocs
 !$ USE OMP_LIB
 
+#ifdef FUTILITY_HAVE_PETSC
+#include <petscversion.h>
+#if ((PETSC_VERSION_MAJOR>=3) && (PETSC_VERSION_MINOR>6))
+USE PETSCSYS, ONLY: PETSC_NULL_CHARACTER
+#endif
+#endif
+
 IMPLICIT NONE
 PRIVATE
 
 #ifdef HAVE_MPI
-
 #ifdef FUTILITY_HAVE_PETSC
-#include <petscversion.h>
 #if ((PETSC_VERSION_MAJOR>=3) && (PETSC_VERSION_MINOR>=6))
 #include <petsc/finclude/petsc.h>
 #else
 #include <finclude/petsc.h>
 #endif
 #undef IS
-PetscErrorCode  :: ierr
-PetscBool :: petsc_isinit
+#if ((PETSC_VERSION_MAJOR>=3) && (PETSC_VERSION_MINOR>6))
+INCLUDE 'mpif.h'
+#endif
 #else
 INCLUDE 'mpif.h'
 #endif
+#endif
```

Comment:
well you include mpif.h in 2 different places... Seems like they can be combined somehow, but maybe I'm wrong.  This si really ugly looking.

Reply:
so...in older versions of PETSc, mpi.h was autoincluded...appears that that may not be the case anymore

Reply:
Alright.  But could we consolidate them?  There are still 2 `#else` branches that `INCLUDE 'mpif.h'`.  Seems like they could be a single one.

Reply:
nope, need two